velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -456,8 +495,14 @@ VectorPtr VectorFuzzer::fuzzFlatPrimitive(
     // First, fill it with random values.
     // TODO: We should bias towards edge cases (min, max, Nan, etc).
     auto kind = vector->typeKind();
-    VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
-        fuzzFlatPrimitiveImpl, kind, vector, rng_, opts_);
+    if (kind == TypeKind::SHORT_DECIMAL) {
```

Comment:
Why do we need these special cases? Why can't we simply provide template specializations for fuzzFlatPrimitiveImpl?

Reply:
There are some specializations not yet implemented for decimals and leads to errors like this:

```
/Users/karteek/workspace/velox/./velox/type/Variant.h:750:47: note: in instantiation of function template specialization 'facebook::velox::VariantConverter::convert<facebook::velox::TypeKind::SHORT_DECIMAL>' requested here
    return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(convert, toKind, value);
                                              ^
/Users/karteek/workspace/velox/./velox/type/Variant.h:279:3: note: candidate constructor not viable: no known conversion from 'typename TypeTraits<(TypeKind)'\f'>::NativeType' (aka 'facebook::velox::UnscaledShortDecimal') to 'const typename detail::VariantTypeTraits<TypeKind::BOOLEAN>::native_type' (aka 'const bool') for 1st argument
  VELOX_VARIANT_SCALAR_MEMBERS(TypeKind::BOOLEAN)
  ^
/Users/karteek/workspace/velox/./velox/type/Variant.h:203:18: note: expanded from macro 'VELOX_VARIANT_SCALAR_MEMBERS'
  /* implicit */ variant(                                             \
                 ^
/Users/karteek/workspace/velox/./velox/type/Variant.h:279:3: note: candidate constructor not viable: no known conversion from 'typename TypeTraits<(TypeKind)'\f'>::NativeType' (aka 'facebook::velox::UnscaledShortDecimal') to 'typename detail::VariantTypeTraits<TypeKind::BOOLEAN>::native_type' (aka 'bool') for 1st argument
  VELOX_VARIANT_SCALAR_MEMBERS(TypeKind::BOOLEAN)
  ^
/Users/karteek/workspace/velox/./velox/type/Variant.h:207:18: note: expanded from macro 'VELOX_VARIANT_SCALAR_MEMBERS'
  /* implicit */ variant(                                             \
                 ^
```

Using `VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL` seems to do the job. There aren't too many references to this macro.